### PR TITLE
refactor(ripple): update for v3

### PR DESCRIPTION
**Pull request by sh7dm**
## Description
Update ripple for Vue 3

## Motivation and Context
Upcoming Vuetify v3

## How Has This Been Tested?
```vue
<template>
  <div v-ripple style="height:200px" class="elevation-2"></div>
</template>
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/directives/ripple/__tests__/ripple.spec.ts [hunk 1]
```diff
@@ -1,69 +1,34 @@
 // Libraries
-import Vue from 'vue'
+import { h, withDirectives, defineComponent, ref, createApp, nextTick, Ref } from 'vue'
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/directives/ripple/__tests__/ripple.spec.ts]:
```suggestion
import { 
  createApp, 
  defineComponent, 
  h, 
  nextTick, 
  Ref,
  ref, 
  withDirectives, 
} from 'vue'
```

File: packages/vuetify/src/directives/ripple/__tests__/ripple.spec.ts [hunk 2]
```diff
@@ -1,69 +1,34 @@
...
-    })
-
-    const wrapper = mount(testComponent, {
-      propsData: {
-        ripple: true,
-      },
-    })
+  it('Ripple should update element property reactively', async () => {
+    const ripple = ref(true)
+    const Test = defineComponent((props: { ripple: Ref<boolean> }) => () => withDirectives(h('div', { class: 'a' }), [ [ Ripple, props.ripple.value ] ]))
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/directives/ripple/__tests__/ripple.spec.ts]: `props: reactive({ ripple: true ])`?

Comment (sh7dm) [reply packages/vuetify/src/directives/ripple/__tests__/ripple.spec.ts]: we don't have `setProps` in Vue 3. `reactive` is great for multiple props, not for 1. I'm also unsure if it works.

Comment (jacekkarczmarczyk) [reply packages/vuetify/src/directives/ripple/__tests__/ripple.spec.ts]: Unfortunately seems that it doesn't work https://codepen.io/jkarczm/pen/MWwgyYJ?editors=1010, however I believe that there's a better way to update component's prop than making prop a ref

And actually I think you don't need props here, this should work as well (didn't check tho):
```ts
const Test = defineComponent(() => () => withDirectives(
  h('div', { class: 'a' }), 
  [[Ripple, ripple.value]],
))
const app = createApp()
const el = document.createElement('div')
app.mount(Test, el)
```

File: packages/vuetify/src/directives/ripple/__tests__/ripple.spec.ts [hunk 3]
```diff
@@ -1,69 +1,42 @@
...
-        }
-        return h('div', data)
-      },
-    })
-
-    const wrapper = mount(testComponent)
+    const Test = defineComponent(() => () => withDirectives(h('div', { class: 'a' }), [ [ Ripple, true ] ]))
+    const app = createApp()
+    const el = document.createElement('div')
+    app.mount(Test, el)
```
Comment (johnleider) [inline packages/vuetify/src/directives/ripple/__tests__/ripple.spec.ts]:
```suggestion

    app.mount(Test, el)
```

File: packages/vuetify/src/directives/ripple/__tests__/ripple.spec.ts [hunk 4]
```diff
@@ -1,69 +1,42 @@
...
-      propsData: {
-        ripple: true,
-      },
-    })
+  it('Ripple should update element property reactively', async () => {
+    const ripple = ref(true)
+    const Test = defineComponent((props: { ripple: Ref<boolean> }) => () => withDirectives(h('div', { class: 'a' }), [ [ Ripple, props.ripple.value ] ]))
+    const app = createApp()
+    const el = document.createElement('div')
+    app.mount(Test, el, { ripple })
```
Comment (johnleider) [inline packages/vuetify/src/directives/ripple/__tests__/ripple.spec.ts]:
```suggestion

    app.mount(Test, el, { ripple })
```

File: packages/vuetify/src/directives/ripple/index.ts [hunk 5]
```diff
@@ -1,8 +1,7 @@
 // Styles
 import './VRipple.sass'
 
-import { VNode, VNodeDirective } from 'vue'
-import { consoleWarn } from '../../util/console'
+import { ObjectDirective, DirectiveBinding } from 'vue'
```
Comment (johnleider) [inline packages/vuetify/src/directives/ripple/index.ts]:
```suggestion
// Setup
import {
  DirectiveBinding,
  ObjectDirective,
} from 'vue'
```

File: packages/vuetify/src/directives/ripple/index.ts [hunk 6]
```diff
@@ -163,7 +162,7 @@ function rippleHide (e: Event) {
   ripples.hide(element)
 }
 
-function updateRipple (el: HTMLElement, binding: VNodeDirective, wasEnabled: boolean) {
+function updateRipple (el: HTMLElement, binding: DirectiveBinding, wasEnabled: boolean) {
   const enabled = isRippleEnabled(binding.value)
   if (!enabled) {
```
Comment (johnleider) [inline packages/vuetify/src/directives/ripple/index.ts]:
```suggestion

  if (!enabled) {
```

File: packages/vuetify/src/directives/ripple/index.ts [hunk 7]
```diff
@@ -163,8 +166,9 @@ function rippleHide (e: Event) {
   ripples.hide(element)
 }
 
-function updateRipple (el: HTMLElement, binding: VNodeDirective, wasEnabled: boolean) {
+function updateRipple (el: HTMLElement, binding: DirectiveBinding, wasEnabled: boolean) {
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/directives/ripple/index.ts]: I'd add RippleDirectiveBinding interface to be in line with other directives

**Review by nekosaur — APPROVED** [review]

**Pull request merged**